projects/stargazer/main.cpp

```diff
@@ -223,7 +239,39 @@ int main(int argc, char* argv[])
         return 1;
     }
 
-    SettingsImpl settings(argc == 2 ? argv[1] : "");
+    std::string path;
+    bool noDaemon(false);
+
+    if (argc == 1)
+        path = "";
+    else
+    {
+        for (int i = 1; i < argc; ++i)
+        {
+            const std::string arg(argv[i]);
+            if (arg == "--help" || arg == "-h")
+            {
+                PrintHelp(argv[0]);
+                return 0;
+            }
+            if (arg == "--version" || arg == "-v")
+            {
+                PrintVersion(argv[0]);
+                return 0;
+            }
+            if (arg == "--foreground" || arg == "-f")
+            {
+                noDaemon = true;
+
+                if (argc == i + 2)
```

Comment:
Why do you need this condition? Wouldn't the following command line argument go into the 'else' branch on the next cycle iteration?